### ISSUE TITLE
Kotlin with Quarkus  - Unable to determine SQL type name for column 

### ISSUE THREAD

**Issue by DawidStuzynski** [vladmihalcea/hypersistence-utils#672]
Hi, I have problem with enums using Kotlin with Quarkus
I made enum like in this article: https://vladmihalcea.com/the-best-way-to-map-an-enum-type-with-jpa-and-hibernate/
and I am getting org.hibernate.MappingException: Unable to determine SQL type name for column 'number' of table 'example_table' error. Application seems to work, during tests I also don't see any errors in console, this only happens when running from main
![image](https://github.com/vladmihalcea/hypersistence-utils/assets/107852058/ccc866af-8ea2-465e-bee9-37403c332390)
is it expected behaviour? 
I made little reproducer of this: https://github.com/DawidStuzynski/enum-imperative/tree/save-enum-warning

**Comment by DawidStuzynski** [comment]
Just after submitting the issue I realized that  @Column(columnDefinition = "number") was missing, with it there is no errors
 @Column(columnDefinition = "number")

**Comment by vladmihalcea** [comment]
@DawidStuzynski If you're using the `PostgreSQLEnumType`, then the column should not be a number, but a PostgreSQL ENUM type. 

Now, for your mapping:

    @Enumerated(EnumType.STRING)
    @Type(PostgreSQLEnumType::class)
    val number: Number 

I don't see why you are mapping it as a `Number` in the JPA entity mapping. It should be a Java Enum. And, on the DB side, it should be like this:

    @Enumerated(EnumType.STRING)
    @Column(columnDefinition = "post_status_info")
    @Type(PostgreSQLEnumType.class)
    private PostStatus status;

where `post_status_info` is:

    CREATE TYPE post_status_info AS ENUM ('PENDING', 'APPROVED', 'SPAM')

Check out [this example](https://github.com/vladmihalcea/hypersistence-utils/blob/cc1c8010f7cc14fbd732521594bf02c8b235972a/hypersistence-utils-hibernate-62/src/test/java/io/hypersistence/utils/hibernate/type/basic/PostgreSQLEnumTest.java#L63) for more details.

In your workaround, you basically switched to using the standard Enum type from HIbernate that uses the ordinal.

**Comment by DawidStuzynski** [comment]
I my example Number enum is postgres type, it is actually misleading. Choosing `Number` as the enum name was not my best idea
My code is looks like: 
```
enum class Number {
    ONE, TWO, THREE
}

CREATE TYPE number AS ENUM (
    'ONE',
    'TWO',
    'THREE'
    );

```


I had project where I used Hyper Persistence utils and all worked fine, during refactoring I removed columnDefinition by accident and all integration tests passed so I din't noticed that something is wrong, after some time I run it from main and noticed some weird logs, so after while I made reproducer in one of my test projects and send an issue. After submitting the issue I looked last time at your article that I was based on, and I realized that columnDefinition is missing (previous test didn't showed me that something is wrong), so I closed issue. Sorry for the confusion
